iceoryx_posh/test/fuzztests/fuzz_helper.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "fuzz_helper.hpp"
+#include "fuzzing.hpp"
+#include "iceoryx_posh/roudi/iceoryx_roudi_components.hpp"
+
+
+std::vector<std::string> FuzzHelper::getStdInMessages() const noexcept
+{
+    std::vector<std::string> stdInMessages;
+    for (std::string line; std::getline(std::cin, line);)
+    {
+        stdInMessages.push_back(line);
+    }
+    return stdInMessages;
+}
+
+std::shared_ptr<RouDiFuzz> FuzzHelper::startRouDiThread() noexcept
+{
```

Comment:
I think this would be cleaner when we create a struct with the components, config and roudi and return it here. Like this here:
```cpp
struct RouDiFuzzThingy {
  iox::roudi::IceOryxRouDiComponents m_rouDiComponents{iox::RouDiConfig_t().setDefaults()};
  RouDiFuzz m_roudi{m_rouDiComponents.rouDiMemoryManager, m_rouDiComponents.portManager};
};

std::unique_ptr<RouDiFuzzThingy> startRouDiThread() noexcept {
  return std::make_unique<RouDiFuzzThingy>();
}
```

Additionally, try to avoid returning a `std::shared_ptr`. The reason is that a `std::unique_ptr` is more efficient and can always be converted to a `std::shared_ptr` if you would like to have one. But you cannot convert a `std::shared_ptr` into a unique ptr.